### PR TITLE
Update stubs for redis-py connection module

### PR DESCRIPTION
**Pull request by vbarbaresi**
used the latest 3.5.3 as reference:
https://github.com/andymccurdy/redis-py/blob/3.5.3/redis/connection.py

Ran stubtest and updated/fixed a few things in this module

### CODE REVIEW

File: third_party/2and3/redis/connection.pyi [hunk 1]
```diff
@@ -48,11 +45,17 @@ class HiredisParser(BaseParser):
...
-    def can_read(self): ...
+    def can_read(self, timeout): ...
+    def read_from_socket(self, timeout=..., raise_on_timeout: bool = ...) -> bool: ...
     def read_response(self): ...
 
 DefaultParser: Any
 
+class Encoder:
+    def __init__(self, encoding, encoding_errors, decode_responses) -> None: ...
+    def encode(self, value: Text) -> bytes: ...
```
Comment (srittau) [inline third_party/2and3/redis/connection.pyi:56]: `encode()` accepts more types: `Text`, `bytes`, `bool`, `float`.

File: third_party/2and3/redis/connection.pyi [hunk 2]
```diff
@@ -83,7 +86,7 @@ class Connection:
         encoding: Text = ...,
         encoding_errors: Text = ...,
         decode_responses: bool = ...,
-        parser_class: Optional[BaseParser] = ...,
+        parser_class: Optional[Type[BaseParser]] = ...,
```
Comment (srittau) [inline third_party/2and3/redis/connection.pyi:89]: This is not `Optional` as far as I can see.

File: third_party/2and3/redis/connection.pyi [hunk 3]
```diff
@@ -125,20 +131,26 @@ class UnixDomainSocketConnection(Connection):
...
+        client_name=...,
     ) -> None: ...
+    def repr_pieces(self) -> List[Tuple[Text, Text]]: ...
+
+def to_bool(value: object) -> bool: ...
 
 class ConnectionPool:
     @classmethod
-    def from_url(cls, url: Text, db: Optional[int] = ..., **kwargs) -> ConnectionPool: ...
+    def from_url(cls, url: Text, db: Optional[int] = ..., decode_components: bool = ..., **kwargs) -> ConnectionPool: ...
```
Comment (srittau) [inline third_party/2and3/redis/connection.pyi:153]: As `decode_components` is keyword-only:

```suggestion
    def from_url(cls, url: Text, db: Optional[int] = ..., *, decode_components: bool = ..., **kwargs) -> ConnectionPool: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, remarks below.

File: third_party/2and3/redis/connection.pyi (hunk 3 above)
Comment (vbarbaresi) [reply third_party/2and3/redis/connection.pyi:153]: Are you sure it's keyword-only?
I get this error in stubtest:
> redis.connection.ConnectionPool.from_url is inconsistent, runtime argument "decode_components" is not keyword-only

https://github.com/andymccurdy/redis-py/blob/3.5.3/redis/connection.py#L950

File: third_party/2and3/redis/connection.pyi (hunk 1 above)
Comment (vbarbaresi) [reply third_party/2and3/redis/connection.pyi]: ✅  updated

File: third_party/2and3/redis/connection.pyi (hunk 2 above)
Comment (vbarbaresi) [reply third_party/2and3/redis/connection.pyi]: right, updated

**Comment by vbarbaresi** [comment]
Mentioning #3544, getting there, little by little

File: third_party/2and3/redis/connection.pyi (hunk 3 above)
Comment (srittau) [reply third_party/2and3/redis/connection.pyi:153]: You are right, I misread this.

**Review by srittau — APPROVED** [review]

**Pull request merged**